src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.cs

```diff
@@ -145,8 +145,10 @@ protected internal override void ClosePin(int pinNumber)
                         _devicePins.Remove(pinNumber);
                     }
 
-                    File.WriteAllText(Path.Combine(GpioBasePath, "unexport"), pinOffset.ToString(CultureInfo.InvariantCulture));
-                    _exportedPins.Remove(pinNumber);
+                    if (_exportedPins.Remove(pinNumber))
```

Comment:
NIt: Add a comment to explain what this if-check is doing.
```suggestion
                    // If this controller wasn't the one that opened the pin, then Remove will return false, so we don't need to close it.
                    if (_exportedPins.Remove(pinNumber))
```